Makefile

```diff
@@ -0,0 +1,123 @@
+#Python3 and virtual environment
+VENV := $(shell mktemp -d)
+PYTHON3 := $(VENV)/bin/python3
+
+#If caramel_tool exists in the venv caramel has been installed
+CARAMEL_TOOL := $(VENV)/bin/caramel_tool
+
+#Terminal formatting
+BOLD := tput bold
+PASS := $(BOLD); tput setaf 2
+FAIL := $(BOLD); tput setaf 1
+LINE := echo "---------------------------------------"
+RESET_TERM := tput op; tput sgr0
+BLR := $(BOLD); $(LINE); $(RESET_TERM) #Bold Line, Reset formatting
+
+#Check for python3 install and virtual environment
+$(PYTHON3):
+	@if [ -z python3 ]; then \
+		$(FAIL);\
+		echo "Python 3 could not be found.";\
+		$(RESET_TERM);\
+		exit 2; \
+	fi
+	python3 -m venv $(VENV)
+
+
+#Install the project via setup.py
+.PHONEY: venv-install
+venv-install: $(CARAMEL_TOOL)
+$(CARAMEL_TOOL): $(PYTHON3) setup.py
+	$(VENV)/bin/pip3 install -e .
+
+
+#Basic tests that caramel can be CARAMEL_TOOL and run with test data, using environment variables for config
+.PHONEY: systest
+systest: export CARAMEL_INI = development.ini
+systest: export CARAMEL_DBURL = sqlite:///caramel.sqlite
+systest: export CARAMEL_LOG_LEVEL = 0
+systest: export CARAMEL_CA_CERT = $(VENV)/caramel.ca.cert
+systest: export CARAMEL_CA_KEY = $(VENV)/caramel.ca.key
+systest: export CARAMEL_CA = 127.0.0.1:6543
+systest: $(CARAMEL_TOOL)
+	@$(BOLD); echo "Removing $(CARAMEL_CA_CERT), $(CARAMEL_CA_KEY) and $(CARAMEL_DBURL)"; $(LINE); $(RESET_TERM)
+	rm -f $(CARAMEL_CA_CERT) $(CARAMEL_CA_KEY) $(CARAMEL_DBURL)
+	@$(BLR)
+
+	@$(BOLD); echo "Generate new CA cert and key with ca_test_input.txt"; $(LINE); $(RESET_TERM)
+	$(VENV)/bin/caramel_ca < ca_test_input.txt
+	@$(BLR)
+
+	@$(BOLD); echo "Create a new DB at $(CARAMEL_DBURL)"; $(LINE); $(RESET_TERM);
+	$(VENV)/bin/caramel_initialize_db
+	@$(BLR)
+
+	@timeout 2s $(VENV)/bin/pserve caramel_launcher.ini \
+		ca_cert=$$CARAMEL_CA_CERT \
+		ca_key=$$CARAMEL_CA_KEY \
+		http_port=6543 \
+		http_host=127.0.0.1 \
+		life_short=48 \
+		life_long=720 \
+		dburl=$$CARAMEL_DBURL \
+		log_level=INFO;\
+		if [ $$? -eq 124 ]; then \
+			$(PASS); $(LINE); echo "Caramel server started and timed out successfully";\
+		else \
+			$(FAIL); $(LINE); echo "System test failed: Caramel server exited before time out with exit code: $$?";\
+			exit 1;\
+		fi;
+	@$(BLR)
+
+	@timeout 2s $(VENV)/bin/caramel_autosign;\
+		if [ $$? -eq 124 ]; then \
+			$(PASS); $(LINE); echo "Autosign server started and timed out successfully";\
+		else \
+			$(FAIL); $(LINE); echo "System test failed: Autosign server exited before time out with exit code: $$?";\
+			exit 1;\
+		fi;
+	@$(BLR)
+
+	@$(BOLD);echo "Start new caramel server and caramel_autosign in the background";$(BLR)
+	setsid $(VENV)/bin/pserve caramel_launcher.ini \
+		ca_cert=$$CARAMEL_CA_CERT \
```

Comment:
Why are these necessary? Don't the tools take the CARAMEL_CA_CERT environment variable?

Reply:
This is not a tool, pserve does not look for our environment variables. But it should probably use the launcher instead, which reads our variables.